packages/terra-form-select/src/shared/_MenuUtil.js

```diff
@@ -126,6 +126,16 @@ class MenuUtil {
     return MenuUtil.flatten(object).find(({ props }) => MenuUtil.isEqual(props.value, value));
   }
 
+  /**
+   * Finds the parent optGroup element matching the requested value.
+   * @param {ReactNode} object - The node being searched.
+   * @param {string} value - The value of the target option.
+   * @return {ReactNode|undefined} - The optGroup. Returns undefined if not found.
+   */
+  static getOptGroupElement(object, value) {
+    return React.Children.toArray(object).find(element => element.type.isOptGroup && element.props.children.find(chidleElement => chidleElement.props.value === value));
```

Comment:
React.Children.toArray() should also be added for `element.props.children` just like single opt-group in form-select there can be single item within opt-group.

Reply:
Updated here edaea09de8e7b5b290819fa74435aa66ddf0e4c8